sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -762,3 +749,124 @@ async def update_tag_properties(
             ),
             repository=repository,
         )
+
+    @distributed_trace_async
+    async def upload_manifest(
+        self, repository: str, manifest: "Union['OCIManifest', IO]", *, tag: "Optional[str]" = None, **kwargs: "Any"
+    ) -> None:
+        """Upload a manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to upload.
+        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str
+        :returns: None
+        :rtype: None
+        """
+        tag_or_digest = tag
+        if tag:
+            stream = manifest
+            if isinstance(manifest, OCIManifest):
+                stream = _serialize_manifest(manifest)
+            tag_or_digest = _compute_digest(stream)
+
+        await self._client.container_registry.create_manifest(
+            repository, tag_or_digest, stream, content_type=OCI_MANIFEST_MEDIA_TYPE, **kwargs)
+
+    @distributed_trace_async
+    async def upload_blob(self, repository, stream, **kwargs):
+        # type: (str, IO, **Any) -> None
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository
+        :param stream: The manifest to upload.
+        :type stream: IO
+        :returns: None
+        :rtype: None
+        """
+        start_upload_response = await self._client.container_registry_blob.start_upload(
+            repository, cls=_return_response, **kwargs)
+        upload_chunk_response = await self._client.container_registry_blob.upload_chunk(
+            start_upload_response['Location'], stream, cls=_return_response, **kwargs)
+        digest = _compute_digest(stream)
+        await self._client.container_registry_blob.complete_upload(
+            digest, upload_chunk_response['Location'], stream, cls=_return_response, **kwargs)
+
+    @distributed_trace_async
+    async def download_manifest(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> ManifestWrapper
+        """Download the manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param str tag_or_digest: The manifest to upload.
+        :returns: ManifestWrapper
+        :rtype: ~container_registry.models.ManifestWrapper
+        """
+        return await self._client.container_registry.get_manifest(
+            repository, tag_or_digest, OCI_MANIFEST_MEDIA_TYPE, **kwargs)
+
+    @distributed_trace_async
+    async def download_blob(self, repository, digest, **kwargs):
+        # type: (str, str, **Any) -> IO | None
+        """Download a blob that is part of an artifact.
+
+        :param str repository: Name of the repository
+        :param str digest: The digest of the blob to download.
+        :returns: IO or None
+        :rtype: IO or None
+        """
+        return await self._client.container_registry_blob.get_blob(repository, digest, **kwargs)
```

Comment:
We're not supporting the `get_digest_from_tag` conversion here?

Reply:
get_blob() consumes the value of digest, not tag. I don't think it's necessary to support tag(include digest value) as input, because users can do the convention.